ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -198,6 +223,20 @@ public void destroyAll(String owner) {
     }
   }
 
+  /**
+   * {@inheritDoc}
+   * Abnormally stopped service may lead to data corruption.
+   * Can take appropriate action by identifying state of service stopped.
+   */
+  @Override
+  public final boolean isClean() {
```

Comment:
Services are started in dependency order. The return of this method is only correct before the service is started. Once started the clean file is deleted and the state will look dirty. This means any service which depends on the `LocalPersistenceService` will check the state of the service on start and always see it as dirty.

Reply:
@chrisdennis 
This scenario is addressed in isClean() API. Please review.